dm412lib.ino

```diff
@@ -99,57 +96,56 @@ public:
         write_channel(level.b);
     }
 
+    void set_light(int light, LightLevels level){
+        lightArray[light] = level;
+    }
+    void write_lights(){
+        for (int i = 0; i < n_lights; i++) {
+            write(lightArray[i]);
+        }
```

Comment:
this is equivalent to
``` python
for light in lightArray:
    write(light)
```
but it only works if you have access to the original array (doesn't work if you pass it by pointer).
```suggestion
        for (auto& light: lightArray) {
            write(light);
        }
```